include/CppUTest/MemoryLeakDetectorMallocMacros.h

```diff
@@ -34,18 +36,11 @@ extern void crash_on_allocation_number(unsigned number);
 
 #endif
 
-/* NOTE on strdup!
- *
- * strdup was implemented earlier, however it is *not* an Standard C function but a POSIX function.
- * Because of that, it can lead to portability issues by providing more than is available on the local platform.
- * For that reason, strdup is *not* implemented as a macro. If you still want to use it, an easy implementation would be:
- *
- * size_t length = 1 + strlen(str);
- * char* result = (char*) cpputest_malloc_location(length, file, line);
- * memcpy(result, str, length);
- * return result;
- *
- */
+#ifdef CPPUTEST_HAVE_STRDUP
+#define strdup(str) cpputest_strdup_location(str, __FILE__, __LINE__)
+#define strndup(str, n) cpputest_strndup_location(str, n, __FILE__, __LINE__)
+#define CPPUTEST_USE_STRDUP_MACROS 1
```

Comment:
The CPPUTEST_USE_STRDUP_MACROS definition was only a matter of stardization in the header and tests, since CPPUTEST_USE_MALLOC_MACROS definition is created below, so I felt it was better to stick with this style. 
In my opinion all these test macros could be refactored/simplified after we get CMake to generate a complete Config.h as Autotools do.

Again, I'd be glad to change it to match your criteria.